src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java

```diff
@@ -77,6 +90,146 @@ default Uni<FindResponse> findDocument(
             });
   }
 
+  /**
+   * This method reads upto system fixed limit
+   *
+   * @param queryExecutor
+   * @param query
+   * @param pageSize
+   * @param objectMapper
+   * @param comparator -
+   * @param numberOfOrderByColumn - Number of order by columns
+   * @param skip - Skip `skip` # of document from the sorted collection before returning the
+   *     documents
+   * @param limit - How many documents to return
+   * @param errorLimit - Count of record on which system to error out, this will be (maximum read
+   *     count for sort + 1)
+   * @return
+   */
+  default Uni<FindResponse> findOrderDocument(
+      QueryExecutor queryExecutor,
+      QueryOuterClass.Query query,
+      int pageSize,
+      ObjectMapper objectMapper,
+      Comparator<ReadDocument> comparator,
+      int numberOfOrderByColumn,
+      int skip,
+      int limit,
+      int errorLimit) {
+    final AtomicInteger documentCounter = new AtomicInteger(0);
+    MinMaxPriorityQueue<ReadDocument> sortedData =
+        MinMaxPriorityQueue.orderedBy(comparator).maximumSize(skip + limit).create();
```

Comment:
but this is now not thread safe, as you are not using it in a reduce way? you have a merge below and this really worries me :warning:

Reply:
The logic doesn't read and write at the same time. sortedData.pool() is done only after the collect() is done. Not sure what the concern here is.